### PR TITLE
Reorganize some cmd/hook plumbing.

### PR DESCRIPTION
**Pull request by spxtr**
I had originally added these channels because it make the picture simpler when `cmd/hook` had plugins written directly in it. Now they're not necessary and only make things confusing.

This cuts some of the duplication required to listen for a new event type.

### CODE REVIEW

**Comment by foxish** [comment]
As I understand it, this change removes some of the serialization of processing which was introduced using channels. Is that perfectly safe? Do we need mutexes guarding any section of the methods which may now be executed concurrently by two separate invocations, such as: handlePullRequestEvent?

**Comment by spxtr** [comment]
It should be safe.

```Go
func (s *Server) handlePullRequestEvent(pr github.PullRequestEvent) {
	l := logrus.WithFields(logrus.Fields{
		"org":    pr.PullRequest.Base.Repo.Owner.Login,
		"repo":   pr.PullRequest.Base.Repo.Name,
		"pr":     pr.Number,
		"author": pr.PullRequest.User.Login,
		"url":    pr.PullRequest.HTMLURL,
	})
	l.Infof("Pull request %s.", pr.Action)
	for p, h := range s.Plugins.PullRequestHandlers(pr.PullRequest.Base.Repo.Owner.Login, pr.PullRequest.Base.Repo.Name) {
		pc := s.Plugins.PluginClient
		pc.Logger = l.WithField("plugin", p)
		if err := h(pc, pr); err != nil {
			pc.Logger.WithError(err).Error("Error handling PullRequestEvent.")
		}
	}
}
```

* `s.Plugins.PullRequestHandlers` is protected by a mutex already.
* `pc := s.Plugins.PluginClient` makes a copy, so the mutation on the next line is only within this function.
* Every element of `PluginClient` is thread-safe, since they're pretty much all `http.Client`s.

That said... I would like to be more certain of this. We shouldn't serialize all plugins though, since some of them can be a little slow. If the CLA plugin needs to wait for API tokens, for instance, it'll take a minute.

**Comment by foxish** [comment]
I agree that we shouldn't serialize everything. But on the flip-side any issues caused by this would be rather hard to debug.
We could inspect some more like you said, to be more certain about this. I'll take a look this evening.

**Comment by spxtr** [comment]
@foxish any ideas? I think this is reasonable and it should be equivalent to what we had before. I guess the idea is that everything in the plugin client must be thread-safe.

**Comment by foxish** [comment]
@spxtr Sorry, I still haven't gotten to looking at it. I'll run through the code once and look at thread safety today.

**Comment by spxtr** [comment]
OK SG, no rush.

File: prow/cmd/hook/events.go [hunk 1]
```diff
@@ -20,40 +20,9 @@ import (
...
-	PullRequestEvents  <-chan github.PullRequestEvent
-	IssueCommentEvents <-chan github.IssueCommentEvent
-	StatusEvents       <-chan github.StatusEvent
-}
-
-// Start starts listening for events. It does not block.
-func (ea *EventAgent) Start() {
-	go func() {
-		for pr := range ea.PullRequestEvents {
-			go ea.handlePullRequestEvent(pr)
```
Comment (spxtr) [inline prow/cmd/hook/events.go]: Actually, they were parallel before, so I don't think I'm introducing any new races. Maybe I just need to document that all of the pluginclient stuff needs to be thread-safe and call it good there?

**Comment by foxish** [comment]
LGTM

File: prow/cmd/hook/events.go (hunk 1 above)
Comment (foxish) [inline prow/cmd/hook/events.go]: Yeah, that should be good. 
The existing plugins don't seem to have any races, so a warning for the new plugins should be fine. If the github client had races, we'd have seen it already.

**Comment by spxtr** [comment]
Thanks :)

**Pull request merged**